lib/index.js

```diff
@@ -123,15 +123,23 @@ internals.Client.prototype.request = function (method, url, options = {}) {
 
 internals.Client.prototype._request = function (method, url, options, relay, _trace) {
 
-    const uri = Url.parse(url);
-
+    const uri = {};
+    let parsedUri;
     if (options.socketPath) {
         uri.socketPath = options.socketPath;
         delete options.socketPath;
+        parsedUri = Url.parse(url);
+    }
+    else {
+        uri.setHost = false;
```

Comment:
Should this impact the code on line 142?

Reply:
So it turns out that in order to maintain compatibility with `Url.parse`, the http request agent will override `host` with `hostname` if both are present. By setting `setHost: false` and manually setting the host header on L42 I was able to preserve the port in the host header. 

This isn't strictly needed, but without it this is a technically breaking change:
```js
const promise = Wreck.request('get', '/foo', { baseUrl: 'http://localhost:0/' });
await expect(promise).to.reject();
expect(promise.req._headers.host).to.equal('localhost:0');
/*Expected 'localhost' to equal specified value: 'localhost:0'*/
```

Similar to the auth prop on the event, I'll let you decide on the importance of preserving that port in the header. Just let me know :)

Reply:
Got it, that makes sense to keep it how you currently have it implemented then.